### ISSUE TITLE
Failing unit tests 50% (CanCreatesNewJSONReader, CanReadJSONReader)

### ISSUE THREAD

**Issue by dariusstrasel** [mechamecho/DataDrivenTesting#5]
Hi Nafissa,

Looks like there are two remaining bugs. It appears that the path is malformed and being interpreted as a UNC path (aka a network file share.) There should be some logic in place to verify the path. 

[System.IO Namespace](https://msdn.microsoft.com/en-us/library/system.io(v=vs.110).aspx) is a good start. Essentially, we should see if this path exists and know whether its a relative or full path. Lastly, we'll want to keep in mind the differences between Windows and Linux for how file paths are handled.

CanCreatesNewJSONReader:
```
Expected: No Exception to be thrown
  But was:  <System.ArgumentException: The UNC path should be of the form \\\\server\\share.
   at System.IO.PathHelper.Normalize(String path, Boolean checkInvalidCharacters, Boolean expandShortPaths)
   at System.IO.Path.GetFullPath(String path)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
   at System.IO.StreamReader..ctor(String path, Encoding encoding, Boolean detectEncodingFromByteOrderMarks, Int32 bufferSize)
   at System.IO.StreamReader..ctor(String path)
   at System.IO.File.OpenText(String path)
   at TestDataAccess.JSONReader.ConvertJSONFileToJObject(JSONFile jsonfile)
   at TestDataAccess.JSONReader.testData(JSONFile jsonFile, String testDataKey, Int32 index)
   at TestDataAccess.JSONReader..ctor(JSONFile jsonFile, String testDataKey, Int32 testDataIndex)
   at TestDataAccess.Tests.JsonFile.CreateJSONReader(JSONFile jsonFile, String testDataKey, Int32 testDataIndex) in C:\Users\siruk\Documents\git\github\DataDrivenTesting\TestDataAccess.Tests\UnitTest1.cs:line 57
   at TestDataAccess.Tests.JsonFile.<>c__DisplayClass5_0.<CanCreatesNewJSONReader>b__0() in C:\Users\siruk\Documents\git\github\DataDrivenTesting\TestDataAccess.Tests\UnitTest1.cs:line 39
   at NUnit.Framework.Constraints.ThrowsConstraint.VoidInvocationDescriptor.Invoke()
   at NUnit.Framework.Constraints.ThrowsConstraint.ExceptionInterceptor.Intercept(Object invocation)>
```

CanReadJSONReader:
```
System.ArgumentException : The UNC path should be of the form \\\\server\\share.
```

**Comment by mechamecho** [comment]
Alright, I will look into it, thank you for the hint!

**Comment by mechamecho** [comment]
Hello Darius, 

I have worked on path validation, could you please take a look at it and tell me if this works the way it should?

Also, I am not so sure on how to approach the Linux vs Windows path differences.

**Comment by dariusstrasel** [comment]
Fixed as of 19c813bd3f92f5acb029ab643377d76b1bdb834

**Comment by dariusstrasel** [comment]
Key thing to think about when considering Linux vs Windows path differences is what they can do uniquely, or things that when used in the other environment will break another. A couple examples:
- Linux file paths are case-sensitive
- Windows can use carets(^) to escape spaces, like^ this
- Windows can use double apostrophes(") to escape spaces, "like this"
- Windows uses backslashes, like\this, and forward-slashes
- Linux uses forward slashes, like/this
**-- Rule of thumb is to use forward slashes since they both support it**
- Linux does not have drive letters, like C:
- Windows treats single backslashes as an escape character, like /n, double slash escapes itself to a single slash like, //

C# lets you create verbatim strings using @, like @"filePath/that/isnt/escaped", which helps in cases where the OS will attempt to change the path. There is also a function to produce the appropriate OS as detailed here:
https://stackoverflow.com/questions/4597572/c-sharp-char-path-separator seperator
